test/functional/tests/fault_injection/test_fault_injection_opencas_load.py

```diff
@@ -0,0 +1,145 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from api.cas import casadm, casadm_parser, cli
+from api.cas.cache_config import CacheMode
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from core.test_run import TestRun
+from test_tools import fs_utils
+from test_tools.dd import Dd
+from test_tools.disk_utils import Filesystem
+from test_utils.filesystem.file import File
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/cas"
+test_file_path = f"{mount_point}/test_file"
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.parametrize("filesystem", Filesystem)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_stop_n_load_cache(cache_mode, filesystem):
+    """
+        title: Fault injection test to check that OpenCAS 'stop -n' option works correctly.
+        description: |
+          Negative test of the ability of the CAS to load unflushed cache
+          when core device is mounted and unmounted.
+        pass_criteria:
+          - No system crash while load cache.
+          - Loading cache without loading metadata fails.
+          - Loading cache with loading metadata finishes with success.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start proper tests"):
+        stop_n_load_cache_fs(cache_mode, filesystem, cache_dev, core_dev)
+        stop_n_load_cache_notfs(cache_mode, cache_dev, core_dev)
+
+
+def stop_n_load_cache_fs(cache_mode, filesystem, cache_dev, core_dev):
+    with TestRun.step("Start cache."):
```

Comment:
Those steps are nested in "with TestRun.step("Start proper tests"):" - it shouldn't be done like that. You can create helpers methods, but they should execute blocks of code without calling steps. Moreover, when all steps are placed in main test's body it is more readable and it would be a lot easier to parse it in the future, e.g. for the documentation purpose.